agent/worker.go

```diff
@@ -42,17 +50,19 @@ type worker struct {
 	db        *bolt.DB
 	executor  exec.Executor
 	listeners map[*statusReporterKey]struct{}
+	secrets   *Secrets
```

Comment:
This doesn't have to be referenced through a pointer from the `worker` struct for its methods to have pointer receivers.